pkg/gds/models/v1/models.go

```diff
@@ -225,6 +225,35 @@ func AppendCertID(vasp *pb.VASP, certID string) (err error) {
 	return nil
 }
 
+// NewCertificate creates and returns a certificate associated with a VASP. This will
+// error if the VASP does not contain an identity certificate.
+func NewCertificate(vasp *pb.VASP, certRequest *CertificateRequest) (cert *Certificate, err error) {
+	// VASP must be not nil.
+	if vasp == nil {
+		return nil, errors.New("must supply a VASP object for certificate creation")
+	}
+
+	// Certificate request must be not nil.
+	if certRequest == nil {
+		return nil, errors.New("must supply a certificate request for certificate creation")
+	}
+
+	// VASP must contain an identity certificate.
+	if vasp.IdentityCertificate == nil {
+		return nil, errors.New("VASP must contain an identity certificate")
+	}
```

Comment:
Not all certificates will be identity certificates - we plan on also issuing sealing certificates that are used for secure envelope PKI. As such I don't think we should rely on the `vasp.IdentityCertificate` for parsing the protocol buffers but should use the same method in the certman code to extract the protocol buffers from the downloaded certificate.